### PR TITLE
PID bugfixes, cleanup and features

### PR DESCRIPTION
**Pull request by eSqadron**
1. Bugfix: Speed control for position control - lock to desired range
2. Feature: Immediate start or soft start setting added
3. Feature: PID refactor - I and D parts added
4. Cleaup: Position calculation macro turned into inline function and cleaned up
5. Cleanup: Degrees per rotation rename
6. Cleanup: Precise encoder interrupts counter modified
7. Cleanup: `is_target_achieved` function minor if-checks optimization
8. Cleanup: `get_control_mode_as_string` removed, as it was unnecessary wrapper for const array

### CODE REVIEW

File: BrushedMotorControllerFirmware/include/calculations.h [hunk 1]
```diff
@@ -3,24 +3,116 @@
...
+			       (int32_t)pid->Ki_denominator));
+	int32_t D = ((int32_t)((int32_t)pid->Kd_numerator * (int32_t)(pid->d_temp - error) /
+			       (int32_t)pid->Kd_denominator));
+	pid->d_temp = error;
+	return P + I + D;
+}
+
+static inline int32_t calculate_pid(int32_t target_speed, int32_t actual_speed, struct PID_def *pid)
+{
+	return calculate_pid_from_error(ERR_C(target_speed, actual_speed), pid);
```
Comment (ghost) [inline BrushedMotorControllerFirmware/include/calculations.h]: if I can see correctly, `ERR_C` is used only here and the name is not really descriptive. You can just put the difference and cast directly In a function call

File: BrushedMotorControllerFirmware/include/calculations.h [hunk 2]
```diff
@@ -3,24 +3,116 @@
...
+/// - position difference (in degrees, int32_t)
+#define CALC_NEW_POS(pos_diff_in_deg, chnl) \
+	(int32_t)(((int32_t)drv_chnls[chnl].curr_pos) + pos_diff_in_deg)
+
+/// Wrap position to range 0-360 degree
+#define WRAP_POS_TO_RANGE(new_pos) \
+	(uint32_t)((new_pos % (int32_t)DEGREES_PER_ROTATION) + \
+		   ((new_pos < 0) ? (int32_t)DEGREES_PER_ROTATION : 0))
+
+#define ABS(v) (v < 0 ? -v : v)
```
Comment (ghost) [inline BrushedMotorControllerFirmware/include/calculations.h]: just include `<stdlib.h>` and use `abs()`

Comment (eSqadron) [reply BrushedMotorControllerFirmware/include/calculations.h]: tried that, abs requires argument to be int, I want to use it with long long int (int64_t)

Comment (ghost) [reply BrushedMotorControllerFirmware/include/calculations.h]: `llabs` should be available as well

Comment (eSqadron) [reply BrushedMotorControllerFirmware/include/calculations.h]: Corrected

File: BrushedMotorControllerFirmware/include/calculations.h (hunk 1 above)
Comment (eSqadron) [reply BrushedMotorControllerFirmware/include/calculations.h]: Corrected

**Review by ghost — APPROVED** [review]

**Pull request merged**